rdmd.d

```diff
@@ -713,9 +725,172 @@ private string[string] getDependencies(string rootModule, string workDir,
         exit(depsExitCode);
     }
 
+    if (!hasDependencies)
+        scope(exit) std.file.remove(rootModule);
+
     return dryRun ? null : readDepsFile();
 }
 
+bool needsInspection(string file)
+{
+    import std.utf : byChar;
+
+    foreach (match; file.matchAll(ctRegex!(`import[^a-zA-Z]*(.*);`, "s")))
```

Comment:
This regex is rather greedy and matches multiple imports at once. For example, `"import foo; import bar;"` is matched as `[["import foo; import bar;", "foo; import bar"]]`. As far as I see, the code below expects one import per match. Exclude ';' from the `(.*)` part.